yesboats command error

```[14:02:36] [Server thread/INFO]: Shinonono_Houki issued server command: /yesboats
[14:02:36] [Server thread/ERROR]: null
org.bukkit.command.CommandException: Unhandled exception executing command 'yesboats' in plugin Yesboats v1.5.1
        at org.bukkit.command.PluginCommand.execute(PluginCommand.java:48) ~[banner-1.20.1-544-server.jar:?]
        at org.bukkit.command.SimpleCommandMap.dispatch(SimpleCommandMap.java:159) ~[banner-1.20.1-544-server.jar:?]
        at org.bukkit.craftbukkit.v1_20_R1.CraftServer.dispatchCommand(CraftServer.java:882) ~[banner-1.20.1-544-server.jar:?]
        at org.bukkit.craftbukkit.v1_20_R1.command.BukkitCommandWrapper.run(BukkitCommandWrapper.java:51) ~[banner-1.20.1-544-server.jar:?]
        at com.mojang.brigadier.CommandDispatcher.execute(CommandDispatcher.java:841) ~[brigadier-1.1.8.jar:?]
        at net.minecraft.commands.Commands.redirect$zbh000$architectury$performCommand(Commands.java:1545) ~[server-intermediary.jar:?]
        at net.minecraft.commands.Commands.performCommand(Commands.java:287) ~[server-intermediary.jar:?]
        at net.minecraft.server.network.ServerGamePacketListenerImpl.performChatCommand(ServerGamePacketListenerImpl.java:3095) ~[server-intermediary.jar:?]
        at net.minecraft.server.network.ServerGamePacketListenerImpl.lambda$handleChatCommand$11(ServerGamePacketListenerImpl.java:1321) ~[server-intermediary.jar:?]
        at net.minecraft.util.thread.BlockableEventLoop.lambda$submitAsync$0(BlockableEventLoop.java:67) ~[server-intermediary.jar:?]
        at java.util.concurrent.CompletableFuture$AsyncSupply.run(CompletableFuture.java:1768) ~[?:?]
        at net.minecraft.server.TickTask.run(TickTask.java:18) ~[server-intermediary.jar:?]
        at net.minecraft.util.thread.BlockableEventLoop.doRunTask(BlockableEventLoop.java:156) ~[server-intermediary.jar:?]
        at net.minecraft.util.thread.ReentrantBlockableEventLoop.doRunTask(ReentrantBlockableEventLoop.java:23) ~[server-intermediary.jar:?]
        at net.minecraft.server.MinecraftServer.doRunTask(MinecraftServer.java:782) ~[server-intermediary.jar:?]
        at net.minecraft.server.MinecraftServer.doRunTask(MinecraftServer.java:164) ~[server-intermediary.jar:?]
        at net.minecraft.util.thread.BlockableEventLoop.pollTask(BlockableEventLoop.java:130) ~[server-intermediary.jar:?]
        at net.minecraft.server.MinecraftServer.pollTaskInternal(MinecraftServer.java:764) ~[server-intermediary.jar:?]
        at net.minecraft.server.MinecraftServer.pollTask(MinecraftServer.java:758) ~[server-intermediary.jar:?]
        at net.minecraft.util.thread.BlockableEventLoop.managedBlock(BlockableEventLoop.java:139) ~[server-intermediary.jar:?]
        at net.minecraft.server.MinecraftServer.waitUntilNextTick(MinecraftServer.java:743) ~[server-intermediary.jar:?]
        at net.minecraft.server.MinecraftServer.runServer(MinecraftServer.java:2145) ~[server-intermediary.jar:?]
        at net.minecraft.server.MinecraftServer.lambda$spin$2(MinecraftServer.java:265) ~[server-intermediary.jar:?]
        at java.lang.Thread.run(Thread.java:842) ~[?:?]
Caused by: java.lang.ArrayIndexOutOfBoundsException: Index 0 out of bounds for length 0
        at rocks.learnercouncil.yesboats.commands.arguments.JoinArg.execute(JoinArg.java:26) ~[?:?]
        at rocks.learnercouncil.yesboats.commands.YesBoatsCmd.onCommand(YesBoatsCmd.java:44) ~[?:?]
        at org.bukkit.command.PluginCommand.execute(PluginCommand.java:46) ~[banner-1.20.1-544-server.jar:?]
        ... 23 more

**To Reproduce**
Steps to reproduce the behavior:
1. /yesboats command error

**Minecraft version information**
- Minecraft Version: 1.20.1
 - Server Software: Banner
 - Server Software Version: Banner-604
 - Plugin Version : Latest Version

--- a/pom.xml
+++ b/pom.xml
@@ -6,7 +6,7 @@
 
     <groupId>rocks.learnercouncil</groupId>
     <artifactId>YesBoats</artifactId>
-    <version>1.5.1</version>
+    <version>1.5.2</version>
     <packaging>jar</packaging>
 
     <name>Yesboats</name>

--- a/src/main/java/rocks/learnercouncil/yesboats/commands/CommandArgument.java
+++ b/src/main/java/rocks/learnercouncil/yesboats/commands/CommandArgument.java
@@ -3,10 +3,40 @@ package rocks.learnercouncil.yesboats.commands;
 import net.md_5.bungee.api.chat.BaseComponent;
 import org.bukkit.command.Command;
 import org.bukkit.command.CommandSender;
+import org.bukkit.util.StringUtil;
+import rocks.learnercouncil.yesboats.YesBoats;
 
+import java.util.ArrayList;
+import java.util.Arrays;
 import java.util.List;
+import java.util.stream.Collectors;
 
 public interface CommandArgument {
+    
+    static BaseComponent[] parseCommand(CommandSender sender, Command cmd, String label, String[] args, CommandArgument[] arguments) {
+        if(args.length < 1) {
+            YesBoats.getInstance().getServer().dispatchCommand(sender, label + " help");
+            return CommandResult.NONE;
+        }
+        for (CommandArgument a : arguments) {
+            final BaseComponent[] result = a.execute(sender, cmd, label, args);
+            if(result.length != 0) {
+                return result;
+            }
+        }
+        //return CommandResult.invalidArgs(label);
+        return CommandResult.NONE;
+    }
+    
+    static List<String> parseTabCompletion(CommandSender sender, Command cmd, String alias, String[] args, CommandArgument[] arguments) {
+        List<String> completions = new ArrayList<>();
+        List<String> argumentCompletions = Arrays.stream(arguments)
+                .flatMap(arg -> arg.tabComplete(sender, cmd, alias, args).stream())
+                .collect(Collectors.toList());
+        if(args.length > 0) StringUtil.copyPartialMatches(args[args.length - 1], argumentCompletions, completions);
+        return completions;
+    }
+        
     BaseComponent[] execute(CommandSender sender, Command cmd, String label, String[] args);
     List<String> tabComplete(CommandSender sender, Command cmd, String alias, String[] args);
 }

--- a/src/main/java/rocks/learnercouncil/yesboats/commands/CommandArgument.java
+++ b/src/main/java/rocks/learnercouncil/yesboats/commands/CommandArgument.java
@@ -24,8 +24,7 @@ public interface CommandArgument {
                 return result;
             }
         }
-        //return CommandResult.invalidArgs(label);
-        return CommandResult.NONE;
+        return CommandResult.invalidArgs(label);
     }
     
     static List<String> parseTabCompletion(CommandSender sender, Command cmd, String alias, String[] args, CommandArgument[] arguments) {

--- a/src/main/java/rocks/learnercouncil/yesboats/commands/CommandResult.java
+++ b/src/main/java/rocks/learnercouncil/yesboats/commands/CommandResult.java
@@ -3,6 +3,7 @@ package rocks.learnercouncil.yesboats.commands;
 
 import net.md_5.bungee.api.ChatColor;
 import net.md_5.bungee.api.chat.BaseComponent;
+import net.md_5.bungee.api.chat.ClickEvent;
 import net.md_5.bungee.api.chat.ComponentBuilder;
 import net.md_5.bungee.api.chat.HoverEvent;
 import net.md_5.bungee.api.chat.hover.content.Text;
@@ -14,7 +15,7 @@ import java.util.function.Consumer;
 import java.util.stream.Collectors;
 
 public class CommandResult {
-
+    
     private static final ChatColor PREFIX = ChatColor.DARK_AQUA;
     private static final ChatColor RESULT = ChatColor.AQUA;
     private static final ChatColor ERROR = ChatColor.RED;
@@ -23,28 +24,26 @@ public class CommandResult {
     private static ComponentBuilder prefix() {
         return new ComponentBuilder().append("[YesBoats] ").color(PREFIX);
     }
+    private static BaseComponent[] result(String message) {
+        return prefix().append(message).color(RESULT).create();
+    }
+    private static BaseComponent[] error(String message) {
+        return prefix().append(message).color(ERROR).create();
+    }
     
     public static final BaseComponent[] NONE = {};
     
-    //Errors
     public static final BaseComponent[]
-            NO_PERMISSION = prefix().append("You don't have permission to execute this command.").color(ERROR).create(),
-            TOO_FEW_ARGS = prefix().append("Too few arguments.").color(ERROR).create(),
-            TOO_MANY_ARGS = prefix().append("Too many arguments").color(ERROR).create(),
-            ARENA_EXISTS = prefix().append("That arena already exists.").color(ERROR).create(),
-            ARENA_NOT_EXIST = prefix().append("That arena doesn't exist.").color(ERROR).create(),
-            PLAYER_NOT_FOUND = prefix().append("Player not found.").color(ERROR).create(),
-            ALREADY_RUNNING = prefix().append("Arena already running.").color(ERROR).create(),
-            NOT_RUNNING = prefix().append("Arena not running.").color(ERROR).create(),
-            TOO_FEW_PLAYERS = prefix().append("Arena doesn't have enough players.").color(ERROR).create(),
-            NOT_IN_ARENA_SELF = prefix().append("You are not in an arena.").color(ERROR).create(),
-            NOT_IN_ARENA_OTHER = prefix().append("That player is not in an arena.").color(ERROR).create(),
-            ALREADY_IN_ARENA_SELF = prefix().append("You are already in an arena.").color(ERROR).create(),
-            ALREADY_IN_ARENA_OTHER = prefix().append("That player is already in an arena.").color(ERROR).create(),
-            INVALID_NUMBER = prefix().append("Invalid number.").color(ERROR).create();
-
-
-    //Results
+            CREATED = result("Arena successfully created. Now editing."),
+            REMOVED = result("Arena successfully removed."),
+            JOINED = result("Joined the game."),
+            LEFT = result("Left the game."),
+            EDITING = result("Now editing."),
+            STARTED = result("Started the game."),
+            STOPPED = result("Stopped the game."),
+            DISPLAYING_PATH = result("Now displaying debug path."),
+            CLEARING_PATH = result("Clearing desplayed debug paths.");
+    
     public static BaseComponent[] joinedOther(String name) {
         return prefix().append("Added ").color(RESULT).append(name).color(SPECIAL).append(" to the game").color(RESULT).create();
     }
@@ -60,17 +59,39 @@ public class CommandResult {
         message.append("====================").color(RESULT);
         return message.create();
     }
-
+    
+    
     public static final BaseComponent[]
-            CREATED = prefix().append("Arena successfully created. Now editing.").color(RESULT).create(),
-            REMOVED = prefix().append("Arena successfully removed.").color(RESULT).create(),
-            JOINED = prefix().append("Joined the game.").color(RESULT).create(),
-            LEFT = prefix().append("Left the game.").color(RESULT).create(),
-            EDITING = prefix().append("Now editing.").color(RESULT).create(),
-            STARTED = prefix().append("Started the game.").color(RESULT).create(),
-            STOPPED = prefix().append("Stopped the game.").color(RESULT).create(),
-            DISPLAYING_PATH = prefix().append("Now displaying debug path.").color(RESULT).create(),
-            CLEARING_PATH = prefix().append("Clearing desplayed debug paths.").color(RESULT).create();
+            NO_PERMISSION = error("You don't have permission to execute this command."),
+            TOO_FEW_ARGS = error("Too few arguments."),
+            TOO_MANY_ARGS = error("Too many arguments"),
+            ARENA_EXISTS = error("That arena already exists."),
+            ARENA_NOT_EXIST = error("That arena doesn't exist."),
+            PLAYER_NOT_FOUND = error("Player not found."),
+            ALREADY_RUNNING = error("Arena already running."),
+            NOT_RUNNING = error("Arena not running."),
+            TOO_FEW_PLAYERS = error("Arena doesn't have enough players."),
+            NOT_IN_ARENA_SELF = error("You are not in an arena."),
+            NOT_IN_ARENA_OTHER = error("That player is not in an arena."),
+            ALREADY_IN_ARENA_SELF = error("You are already in an arena."),
+            ALREADY_IN_ARENA_OTHER = error("That player is already in an arena.");
+
+    public static BaseComponent[] needsPlayer(String label) {
+        return prefix().append("Command '").color(ERROR)
+                .append('/' + label).color(SPECIAL)
+                .append("' must be executed by a player.").color(ERROR)
+                .create();
+    }
+    public static BaseComponent[] invalidArgs(String label) {
+        return prefix().append("Invalid arguments. Try '").color(ERROR)
+                .append('/' + label + " help").color(SPECIAL).event(new ClickEvent(ClickEvent.Action.RUN_COMMAND, '/' + label + " help"))
+                .append("' for help.")
+                .create();
+    }
+    public static BaseComponent[] invalidNumber(String number) {
+        return prefix().append("'").color(ERROR).append(number).color(SPECIAL).append("' is not a valid number.").color(ERROR).create();
+    }
+    
     
     public static BaseComponent[] getHelpMenu(String label, CommandSender sender) {
         return new HelpMenuBuilder(label, sender)

--- a/src/main/java/rocks/learnercouncil/yesboats/commands/CommandResult.java
+++ b/src/main/java/rocks/learnercouncil/yesboats/commands/CommandResult.java
@@ -16,10 +16,11 @@ import java.util.stream.Collectors;
 
 public class CommandResult {
     
-    private static final ChatColor PREFIX = ChatColor.DARK_AQUA;
-    private static final ChatColor RESULT = ChatColor.AQUA;
-    private static final ChatColor ERROR = ChatColor.RED;
-    private static final ChatColor SPECIAL = ChatColor.YELLOW;
+    private static final ChatColor
+            PREFIX = ChatColor.DARK_AQUA,
+            RESULT = ChatColor.AQUA,
+            ERROR = ChatColor.RED,
+            SPECIAL = ChatColor.YELLOW;
     
     private static ComponentBuilder prefix() {
         return new ComponentBuilder().append("[YesBoats] ").color(PREFIX);
@@ -85,7 +86,7 @@ public class CommandResult {
     public static BaseComponent[] invalidArgs(String label) {
         return prefix().append("Invalid arguments. Try '").color(ERROR)
                 .append('/' + label + " help").color(SPECIAL).event(new ClickEvent(ClickEvent.Action.RUN_COMMAND, '/' + label + " help"))
-                .append("' for help.")
+                .append("' for help.").color(ERROR)
                 .create();
     }
     public static BaseComponent[] invalidNumber(String number) {

--- a/src/main/java/rocks/learnercouncil/yesboats/commands/YesBoatsCmd.java
+++ b/src/main/java/rocks/learnercouncil/yesboats/commands/YesBoatsCmd.java
@@ -1,17 +1,15 @@
 package rocks.learnercouncil.yesboats.commands;
 
-import net.md_5.bungee.api.chat.BaseComponent;
 import org.bukkit.command.Command;
 import org.bukkit.command.CommandSender;
 import org.bukkit.command.TabExecutor;
 import org.bukkit.entity.Player;
-import org.bukkit.util.StringUtil;
 import org.jetbrains.annotations.NotNull;
 import rocks.learnercouncil.yesboats.commands.arguments.*;
 
 import java.util.ArrayList;
+import java.util.Collections;
 import java.util.List;
-import java.util.stream.Collectors;
 
 public class YesBoatsCmd implements TabExecutor {
 
@@ -35,29 +33,22 @@ public class YesBoatsCmd implements TabExecutor {
 
     @Override
     public boolean onCommand(@NotNull CommandSender sender, @NotNull Command cmd, @NotNull String label, String[] args) {
+        if(!cmd.getName().equalsIgnoreCase("yesboats")) return false;
         if(!(sender instanceof Player)) {
             sender.sendMessage("[YesBoats] This command must be executed by a player");
             return true;
         }
-        if(cmd.getName().equalsIgnoreCase("yesboats")) {
-            for (CommandArgument a : arguments) {
-                final BaseComponent[] result = a.execute(sender, cmd, label, args);
-                if(result.length != 0) {
-                    sender.spigot().sendMessage(result);
-                    return true;
-                }
-            }
+        if(!sender.hasPermission("yesboats.commands.yesboats.user")) {
+            sender.spigot().sendMessage(CommandResult.NO_PERMISSION);
         }
-        return false;
+        
+        sender.spigot().sendMessage(CommandArgument.parseCommand(sender, cmd, label, args, arguments.toArray(CommandArgument[]::new)));
+        return true;
     }
 
     @Override
     public List<String> onTabComplete(@NotNull CommandSender sender, @NotNull Command cmd, @NotNull String alias, String[] args) {
-        List<String> arguments;
-        List<String> completions = new ArrayList<>();
-        arguments = this.arguments.stream().flatMap(arg -> arg.tabComplete(sender, cmd, alias, args).stream()).collect(Collectors.toList());
-        StringUtil.copyPartialMatches(args[args.length - 1], arguments, completions);
-        return completions;
-
+        if(!sender.hasPermission("yesboats.commands.yesboats.user")) return Collections.emptyList();
+        return CommandArgument.parseTabCompletion(sender, cmd, alias, args, arguments.toArray(CommandArgument[]::new));
     }
 }

--- a/src/main/java/rocks/learnercouncil/yesboats/commands/YesBoatsCmd.java
+++ b/src/main/java/rocks/learnercouncil/yesboats/commands/YesBoatsCmd.java
@@ -35,11 +35,12 @@ public class YesBoatsCmd implements TabExecutor {
     public boolean onCommand(@NotNull CommandSender sender, @NotNull Command cmd, @NotNull String label, String[] args) {
         if(!cmd.getName().equalsIgnoreCase("yesboats")) return false;
         if(!(sender instanceof Player)) {
-            sender.sendMessage("[YesBoats] This command must be executed by a player");
+            sender.spigot().sendMessage(CommandResult.needsPlayer(label));
             return true;
         }
         if(!sender.hasPermission("yesboats.commands.yesboats.user")) {
             sender.spigot().sendMessage(CommandResult.NO_PERMISSION);
+            return true;
         }
         
         sender.spigot().sendMessage(CommandArgument.parseCommand(sender, cmd, label, args, arguments.toArray(CommandArgument[]::new)));

--- a/src/main/java/rocks/learnercouncil/yesboats/commands/YesBoatsCmd.java
+++ b/src/main/java/rocks/learnercouncil/yesboats/commands/YesBoatsCmd.java
@@ -7,29 +7,21 @@ import org.bukkit.entity.Player;
 import org.jetbrains.annotations.NotNull;
 import rocks.learnercouncil.yesboats.commands.arguments.*;
 
-import java.util.ArrayList;
 import java.util.Collections;
 import java.util.List;
 
 public class YesBoatsCmd implements TabExecutor {
 
-    private final List<CommandArgument> arguments = initializeArgs();
-
-    private List<CommandArgument> initializeArgs() {
-        List<CommandArgument> result = new ArrayList<>();
-
-        result.add(new JoinArg());
-        result.add(new LeaveArg());
-        result.add(new AddArg());
-        result.add(new RemoveArg());
-        result.add(new EditArg());
-        result.add(new StartArg());
-        result.add(new StopArg());
-        result.add(new ReportsArg());
-        result.add(new HelpArg());
-
-        return result;
-    }
+    private final CommandArgument[] ARGUMENTS = {
+            new JoinArg(),
+            new LeaveArg(),
+            new AddArg(),
+            new RemoveArg(),
+            new EditArg(),
+            new StopArg(),
+            new ReportsArg(),
+            new HelpArg(),
+    };
 
     @Override
     public boolean onCommand(@NotNull CommandSender sender, @NotNull Command cmd, @NotNull String label, String[] args) {
@@ -43,13 +35,13 @@ public class YesBoatsCmd implements TabExecutor {
             return true;
         }
         
-        sender.spigot().sendMessage(CommandArgument.parseCommand(sender, cmd, label, args, arguments.toArray(CommandArgument[]::new)));
+        sender.spigot().sendMessage(CommandArgument.parseCommand(sender, cmd, label, args, ARGUMENTS));
         return true;
     }
 
     @Override
     public List<String> onTabComplete(@NotNull CommandSender sender, @NotNull Command cmd, @NotNull String alias, String[] args) {
         if(!sender.hasPermission("yesboats.commands.yesboats.user")) return Collections.emptyList();
-        return CommandArgument.parseTabCompletion(sender, cmd, alias, args, arguments.toArray(CommandArgument[]::new));
+        return CommandArgument.parseTabCompletion(sender, cmd, alias, args, ARGUMENTS);
     }
 }

--- a/src/main/java/rocks/learnercouncil/yesboats/commands/arguments/ReportsArg.java
+++ b/src/main/java/rocks/learnercouncil/yesboats/commands/arguments/ReportsArg.java
@@ -42,7 +42,7 @@ public class ReportsArg implements CommandArgument {
                 DebugPath.display(index, (Player) sender);
                 return CommandResult.DISPLAYING_PATH;
             } catch (NumberFormatException e) {
-                return CommandResult.INVALID_NUMBER;
+                return CommandResult.invalidNumber(args[2]);
             }
         }
         return TOO_MANY_ARGS;

